src/api/source/sql.js

```diff
@@ -68,7 +68,7 @@ export default class SQL extends BaseWindshaft {
         if (query === '') {
             throw new CartoValidationError('source', 'nonValidQuery');
         }
-        var sqlRegex = /(SELECT|select)\s+.*\s+(FROM|from)\s+.*/;
+        var sqlRegex = /\bSELECT\b/i;
```

Comment:
If the user is a `lowercase`fan this could be not enough, maybe `\b(SELECT|select)\b`?

Reply:
The `i` is for case insensitive, so it should match SELECT, select as well as Select or sElEct

Reply:
Ah, you are right, I missed that part!